core/src/main/java/com/datastax/oss/driver/api/core/config/OptionsMap.java

```diff
@@ -381,6 +381,8 @@ protected static void fillWithDriverDefaults(OptionsMap map) {
     map.put(TypedDriverOption.LOAD_BALANCING_DC_FAILOVER_MAX_NODES_PER_REMOTE_DC, 0);
     map.put(TypedDriverOption.LOAD_BALANCING_DC_FAILOVER_ALLOW_FOR_LOCAL_CONSISTENCY_LEVELS, false);
     map.put(TypedDriverOption.METRICS_GENERATE_AGGREGABLE_HISTOGRAMS, true);
+    map.put(
+        TypedDriverOption.LOAD_BALANCING_DC_FAILOVER_PREFERRED_REMOTE_DCS, ImmutableList.of(""));
```

Comment:
Hi @nitinitt . Thanks for this contribution, very useful.
While testing it on my side, I've stumbled upon this default (we use OptionsMap for configuring the driver). With this default, we never go to [`buildRemoteQueryPlanAll`](https://github.com/apache/cassandra-java-driver/pull/1896/files#diff-bbdb6181132c9a28e06a98058d5b3db08d3398ffb4233f242f8da44b26a9249bR333-R334) and fall into `buildRemoteQueryPlanPreferred`, which obviously cannot find nodes of the DC `""`.
Was this set intentionally to default to a Set with empty string for some reason?

Reply:
Hi @jahstreet 
Is the code empty check: https://github.com/apache/cassandra-java-driver/pull/1896/files#diff-bbdb6181132c9a28e06a98058d5b3db08d3398ffb4233f242f8da44b26a9249bR333-R336 
`if (preferredRemoteDcs.isEmpty()) {
      return new CompositeQueryPlan(local, buildRemoteQueryPlanAll());
    }`
not getting invoked?

Reply:
I actually doesn't, cause there is an empty string in the array.